pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/plugin/internal/ValidationPlugin.java

```diff
@@ -56,13 +56,18 @@ public void apply(final Project project) {
         project.getTasks().withType(AbstractPythonTestSourceDefaultTask.class,
             task -> task.dependsOn(project.getTasks().getByName(TASK_INSTALL_PROJECT.getValue())));
 
+        CoverageExtension cov = ExtensionUtils.maybeCreate(project, "coverage", CoverageExtension.class);
         /*
          * Run tests using py.test.
          *
          * This uses the ``setup.cfg`` if present to configure py.test.
+         * Pytest task will not be run if coverage is set to true, unless explicitly executed. If coverage is enabled,
+         * the coverage task will execute the test suite.
          */
         project.getTasks().create(TASK_PYTEST.getValue(), PyTestTask.class,
-            task -> task.onlyIf(it -> project.file(settings.testDir).exists()));
+            task -> task.onlyIf(it -> project.file(settings.testDir).exists()
+                && (!cov.isRun() || project.getGradle().getStartParameter().getTaskNames().contains(TASK_PYTEST.getValue()))
```

Comment:
`Task#onlyIf` with a lambda is a great pattern.

Now I suggest extracting the boolean return value to a variable, then log the value using `LOG` (so that we know why pytest is being skipped!), and finally return the extracted variable containing the value.

Something like:
```
project.getTasks().create(TASK_PYTEST.getValue(), PyTestTask.class, 
  task -> task.onlyIf(it -> {
    boolean isRunPytest = project.file(settings.testDir).exists() &&
    (!cov.isRun() || project.getGradle().getStartParameter().getTaskNames().contains(TASK_PYTEST.getValue());
  if(!isRunPytest) {
    LOG.info("Not running {} because of reasons", getPath();
  }
  return isRunPytest;
}):
```